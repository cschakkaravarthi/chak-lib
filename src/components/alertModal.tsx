import React, { FC } from 'react';
import { Modal, Row, Col } from 'react-bootstrap';
import { UploadButton } from './uploadButton';

import '../../scss/_modal.scss';

export interface AlertModalProps {
  show: boolean;
  dialogMessage: string;
  alertButtonText?: string;
  handleClose: () => void;
}

export const AlertModal: FC<AlertModalProps> = props => {
  const {
    show,
    dialogMessage,
    alertButtonText = 'Ok',
    handleClose
  } = props;
  return (
    <Modal show={show} onHide={handleClose} className="alert-modal">
      <Modal.Body className="text-condensed font-weight-bold text-center pt-5">{dialogMessage}</Modal.Body>
      <Modal.Footer className="border-0 d-flex justify-content-center">
        <Row className="w-100 mb-5">
          <Col lg="12" md="12" className="text-center">
            <UploadButton
              handleClick={handleClose}
              title={alertButtonText}
              uniqueIdentifier="close"
              className="font-size-sm text-upper primary-button"
            />
          </Col>
        </Row>
      </Modal.Footer>
    </Modal>
  );
};
