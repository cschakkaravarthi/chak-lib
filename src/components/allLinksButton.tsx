import React, { FC } from 'react';
import { UIButtonProps } from '../interfaces';

export const AllLinksButton: FC<UIButtonProps> = props => {
  return (
    <button
      onClick={props.handleClick}
      className="d-inline-flex align-items-center align-middle btn-upper dark-background text-white"
    >
      <img
        src={props.imageSrc}
        className='d-md-inline order-md-2 all-links-image'
      />
      <span className="font-size-sm m-3">All Links</span>
    </button>
  );
};
