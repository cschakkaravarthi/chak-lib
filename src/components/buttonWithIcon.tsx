import React, { FC } from 'react';
import { UIButtonProps } from '../interfaces';
import { Image } from 'react-bootstrap';

export interface ButtonWithIconProps extends UIButtonProps {
  imageSrc?: string;
  imageWidth?: number | string;
  imageHeight?: number | string;
  labelClassName?: string;
}

export const ButtonWithIcon: FC<ButtonWithIconProps> = props => {
  const {
    className,
    uniqueIdentifier,
    handleClick,
    title,
    imageSrc,
    imageWidth = 20,
    imageHeight = 20,
    labelClassName = 'gray-600',
    hrefLink
  } = props;
  return (
    <a
      className={`font-size-sm cursor-pointer ${className}`}
      key={uniqueIdentifier}
      onClick={handleClick}
      href={hrefLink}
    >
      {imageSrc && <Image src={imageSrc} width={imageWidth} height={imageHeight} />}
      <span className={`pl-2 ${labelClassName}`}>{title}</span>
    </a>
  );
};
