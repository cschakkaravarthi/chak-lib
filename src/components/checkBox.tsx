import React, { FC } from 'react';
import { UIButtonProps } from '../interfaces';
import { Form } from 'react-bootstrap';

import '../../scss/_checkbox.scss';

export const Checkbox: FC<UIButtonProps> = props => {
  return (
    <Form onClick={props.handleClick}>
      <Form.Check
        custom
        type="checkbox"
        id={props.uniqueIdentifier}
        label={props.title}
        checked={props.checked}
      />
    </Form>
  );
};
