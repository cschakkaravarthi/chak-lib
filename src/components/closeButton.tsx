import React, { FC } from 'react';
import { UIButtonProps } from '../interfaces';
import { Button } from 'react-bootstrap';
import { images } from '../assets/images';

export const CloseButton: FC<UIButtonProps> = props => {
  return (
    <Button
      variant="dark"
      className="dark-background p-3 menu-toggler"
      onClick={props.handleClick}
    >
      <img src={images.close} width={50} height={50} />
    </Button>
  );
};
