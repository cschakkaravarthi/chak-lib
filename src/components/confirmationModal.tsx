import React, { FC } from 'react';
import { Modal, Row, Col } from 'react-bootstrap';
import { UploadButton } from './uploadButton';

import '../../scss/_modal.scss';

export interface ConfirmationModalProps {
  show: boolean;
  dialogMessage: string;
  okayButtonText?: string;
  cancelButtonText?: string;
  handleClose: () => void;
  handleSave: () => void;
}

export const ConfirmationModal: FC<ConfirmationModalProps> = props => {
  const {
    show,
    dialogMessage,
    okayButtonText = 'Okay',
    cancelButtonText = 'Cancel',
    handleClose,
    handleSave
  } = props;
  return (
    <Modal show={show} onHide={handleClose} className="confirmation-modal">
      <Modal.Body className="text-condensed font-weight-bold text-center pt-5">{dialogMessage}</Modal.Body>
      <Modal.Footer className="border-0 d-flex justify-content-center">
        <Row className="w-100 mb-5">
          <Col xs="6" className="text-center">
            <UploadButton
              handleClick={ handleClose }
              title={cancelButtonText}
              uniqueIdentifier="close"
              className="font-size-sm text-upper"
            />
          </Col>
          <Col xs="6" className="text-center">
            <UploadButton
              handleClick={handleSave}
              title={okayButtonText}
              uniqueIdentifier="save"
              className="font-size-sm text-upper"
            />
          </Col>
        </Row>
      </Modal.Footer>
    </Modal>
  );
};
