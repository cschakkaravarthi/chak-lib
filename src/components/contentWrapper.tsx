import React, { FC } from 'react';
import SourceCard from 'react-bootstrap/Card';

// In the content wrapper you will need to make a skeleton appear only
// in the wrapper. Not in the children. Children already has it's own skeleton.
import Skeleton from 'react-loading-skeleton';

export interface ContentWrapperProps {
  title: string;
  subtitle?: string;
  bodyClass?: string;
  topBorder?: string;
  titleClass?: string;
  isFetching?: boolean;
  wrapperClass?: string;
  subtitleClass?: string;
  rightCornerComponentText?: string;
  RightCornerComponent?: React.ComponentType;
  // if true, it will show an optional title on the left and a right corner
  // optional text set by the props rightCornerComponentText and RightCornerComponent.
  showContentTop?: boolean;
}

export const ContentWrapper: FC<ContentWrapperProps> = props => {
  const {
    title,
    children,
    subtitle,
    bodyClass,
    wrapperClass,
    isFetching = false,
    RightCornerComponent,
    showContentTop = true,
    topBorder = 'lightblue',
    rightCornerComponentText,
    subtitleClass = 'gray-600 font-size-sm',
    titleClass = 'font-size-xl font-weight-bold'
  } = props;

  return (
    <SourceCard className={`${wrapperClass} content-wrapper-top-border-${topBorder} rounded-0 border-0`}>
      <SourceCard.Body className={bodyClass}>
        {showContentTop ? (
          <div className="row">
            <div className={RightCornerComponent ? 'col-9' : 'col-12'}>
              {title && (
                <SourceCard.Title className={titleClass}>
                  {!isFetching ? title : <Skeleton width={200} />}
                </SourceCard.Title>
              )}
              {subtitle && <span className={subtitleClass}>{!isFetching ? subtitle : <Skeleton width={125} />}</span>}
            </div>
            {RightCornerComponent && (
              <div className="col-3">
                <RightCornerComponent>
                  <u>{!isFetching ? rightCornerComponentText : <Skeleton width={100} />}</u>
                </RightCornerComponent>
              </div>
            )}
          </div>
        ) : null}
        {children}
      </SourceCard.Body>
    </SourceCard>
  );
};
