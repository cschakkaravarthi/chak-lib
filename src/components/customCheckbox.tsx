import React, { FC } from 'react';
import { UIButtonProps } from '../interfaces';

import '../../scss/_checkbox.scss';

export const CustomCheckbox: FC<UIButtonProps> = props => {
  return (
    <div className={`left-label-checkbox ${props.className}`} onClick={props.handleClick}>
      <label className={props.checked ? 'checked' : ''}>{props.title}</label>
      <input
        name={props.name}
        type="checkbox"
        id={props.uniqueIdentifier}
        checked={props.checked}
      />
    </div>
  );
};
