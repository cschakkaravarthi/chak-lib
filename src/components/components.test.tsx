import React from 'react';

import { shallowRender } from '../services/testHelper';

import { AllLinksButton } from './allLinksButton';
import { HamburgerButton } from './hamburgerButton';
import { NotificationsBadge } from './notificationsBadge';
import { SiteLogo } from './siteLogo';
import { UserBadge } from './userBadge';
import { ContentWrapper } from './contentWrapper';
import { PrimaryButton } from './primaryButton';
import { SecondaryButton } from './secondaryButton';
import { CloseButton } from './closeButton';
import { UploadButton } from './uploadButton';
import { Checkbox } from './checkBox';
import { EditButton } from './editButton';
import { FooterFeedbackView } from './footerFeedbackView';
import { images } from '../assets/images';
import { ThumbButton } from './thumbButton';
import { OrgChart } from './orgChart';
import { dummyUsers } from './../seeds/OrgChart.model';
import { CustomCheckbox } from './customCheckbox';
import { ButtonWithIcon } from './buttonWithIcon';
import { Toast } from './toast';
import { Error } from './error';
import { ConfirmationModal } from './confirmationModal';
import { AlertModal } from './alertModal';
import { DateRangeHeader } from './dateRangeHeader';
import { Partition } from './partition';
import { Textarea } from './textarea';

describe('Components', () => {
  it('Textarea should render correctly', () => {
    const tree = shallowRender(
      <Textarea
        label="Name"
        maxLength={400}
        controlId="formName"
        subLabelClass="ml-3"
        subLabel="Maximun 400 characters"
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('Date Range Header should render correctly', () => {
    const tree = shallowRender(<DateRangeHeader leftTitle="From" rightTitle="Date" />);
    expect(tree).toMatchSnapshot();
  });

  it('ContentWrapper should render correctly', () => {
    const tree = shallowRender(
      <ContentWrapper title="Test title">
        <p>Testing ContentWrapper children prop</p>
      </ContentWrapper>
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly for AllLinksButton', () => {
    const tree = shallowRender(
      <AllLinksButton
        handleClick={() => {
          alert('clicked');
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly for HambugerButton', () => {
    const tree = shallowRender(
      <HamburgerButton
        handleClick={() => {
          alert('clicked');
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly for NotificationsBadge', () => {
    const tree = shallowRender(
      <NotificationsBadge
        handleClick={() => {
          alert('clicked');
        }}
        notificationCount={3}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly for SiteLogo', () => {
    const tree = shallowRender(<SiteLogo logoSrc={images.siteLogo} handleClick={() => alert('clicked')} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly for UserBadge', () => {
    const tree = shallowRender(
      <UserBadge
        fullName="Luz Lacson"
        handleClick={() => alert('clicked')}
        imageUrl="https://media.licdn.com/dms/image/C4D03AQHgxarEg1Cvuw/profile-displayphoto-shrink_800_800/0?e=1575504000&v=beta&t=thBaLX-J3-MoHdpJ3MvombNd4LOKfMHNRPLlmopM6KI"
      />
    );
    expect(tree).toMatchSnapshot();
  });
});

it('should render correctly for Primary button', () => {
  const tree = shallowRender(
    <PrimaryButton handleClick={() => alert('Primary Button clicked')} title="Primary Button" />
  );
  expect(tree).toMatchSnapshot();
});

it('should render correctly for Secondary Button', () => {
  const tree = shallowRender(
    <SecondaryButton handleClick={() => alert('Secondary Button clicked')} title="Secondary Button" />
  );
  expect(tree).toMatchSnapshot();
});

it('should render correctly for Close toggle Button', () => {
  const tree = shallowRender(
    <CloseButton
      handleClick={() => {
        alert('clicked');
      }}
    />
  );
  expect(tree).toMatchSnapshot();
});

it('should render correctly for CheckBox', () => {
  const tree = shallowRender(
    <Checkbox
      handleClick={() => {
        alert('clicked');
      }}
      title="CheckLabel"
    />
  );
  expect(tree).toMatchSnapshot();
});

it('should render correctly for Edit Button', () => {
  const tree = shallowRender(
    <EditButton
      handleClick={() => {
        alert('clicked');
      }}
      title="Edit"
      hrefLink="https://www.google.com"
    />
  );
  expect(tree).toMatchSnapshot();
});

it('should render correctly for Upload Button', () => {
  const tree = shallowRender(
    <UploadButton
      handleClick={() => {
        alert('clicked');
      }}
      title="Upload"
      uniqueIdentifier="1"
    />
  );
  expect(tree).toMatchSnapshot();
});

it('should render correctly for FooterFeedbackView', () => {
  const tree = shallowRender(
    <FooterFeedbackView
      handleClick={() => {
        alert('clicked');
      }}
      title="Edit"
    />
  );
  expect(tree).toMatchSnapshot();
});

it('should render correctly for Thumb button unlike', () => {
  const tree = shallowRender(
    <ThumbButton
      liked={false}
      numberOfLikes={12}
      handleClick={() => {
        alert('clicked');
      }}
    />
  );
  expect(tree).toMatchSnapshot();
});

it('should render correctly for Thumb button like', () => {
  const tree = shallowRender(
    <ThumbButton
      liked={true}
      numberOfLikes={12}
      handleClick={() => {
        alert('clicked');
      }}
    />
  );
  expect(tree).toMatchSnapshot();
});

it('should render correctly for Organisation Chart', () => {
  const tree = shallowRender(<OrgChart Users={dummyUsers} handleManagerClick={() => console.log('clicked')} />);
  expect(tree).toMatchSnapshot();
});

it('should render correctly for CheckBox', () => {
  const tree = shallowRender(
    <CustomCheckbox
      handleClick={() => {
        alert('clicked');
      }}
      title="label"
      checked={true}
      className="font-size-sm pr-1"
    />
  );
  expect(tree).toMatchSnapshot();
});

it('should render correctly for Button with Icon', () => {
  const tree = shallowRender(
    <ButtonWithIcon
      title="Add to My Contacts"
      handleClick={() => {
        alert('clicked');
      }}
      imageSrc={images.addToContactsIcon}
      imageHeight={20}
      imageWidth={20}
    />
  );
  expect(tree).toMatchSnapshot();
});

it('should render correctly for Toast', () => {
  const tree = shallowRender(<Toast message="Added to My Contacts" variant="success" className="show" />);
  expect(tree).toMatchSnapshot();
});

it('should render correctly for Toast', () => {
  const tree = shallowRender(<Toast message="Contact is already in My Contacts" variant="error" className="show" />);
  expect(tree).toMatchSnapshot();
});

it('should render correctly for Error', () => {
  const tree = shallowRender(
    <Error
      backgroundImage={images.faceBrowser}
      title="Please upgrade your browser."
      message="UMG Central has detected that you are using an unsupported browser. Please use links below to download and install a supported browser."
      isShowBrowsers={true}
    />
  );
  expect(tree).toMatchSnapshot();
});

it('should render correctly for Error Restricted', () => {
  const tree = shallowRender(
    <Error
      backgroundImage={images.faceLockNew}
      title="Restricted Content"
      message="To access this content, please connect to the UMG Network via VPN."
      isShowBrowsers={false}
      urlText="View vpn instructions"
      urlLink="https://central.umusic.com/pages/315"
    />
  );
  expect(tree).toMatchSnapshot();
});

it('should render correctly for Confirmation Modal', () => {
  const tree = shallowRender(
    <ConfirmationModal
      dialogMessage="Are you sure you want to remove this contact?"
      show={true}
      okayButtonText="Remove"
      handleClose={() => alert('clicked close')}
      handleSave={() => alert('clicked save')}
    />
  );
  expect(tree).toMatchSnapshot();
});

it('should render correctly for Alert Modal', () => {
  const tree = shallowRender(
    <AlertModal
      dialogMessage="Please enter both month and day for Birthday."
      show={true}
      alertButtonText="Ok"
      handleClose={() => alert('clicked close')}
    />
  );
  expect(tree).toMatchSnapshot();
});

it('should render correctly for Partition with message', () => {
  const tree = shallowRender(<Partition message="The information below is private" />);
  expect(tree).toMatchSnapshot();
});
