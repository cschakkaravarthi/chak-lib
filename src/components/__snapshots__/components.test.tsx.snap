// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components ContentWrapper should render correctly 1`] = `
<ForwardRef
  body={false}
  className="undefined content-wrapper-top-border-lightblue rounded-0 border-0"
>
  <ForwardRef>
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <ForwardRef
          className="font-size-xl font-weight-bold"
        >
          Test title
        </ForwardRef>
      </div>
    </div>
    <p>
      Testing ContentWrapper children prop
    </p>
  </ForwardRef>
</ForwardRef>
`;

exports[`Components Date Range Header should render correctly 1`] = `
<div
  className="date-range-header d-flex justify-content-between align-items-baseline undefined"
>
  <p>
    From
  </p>
  <p>
    Date
  </p>
</div>
`;

exports[`Components Textarea should render correctly 1`] = `
<ForwardRef
  controlId="formName"
>
  <ForwardRef
    column={false}
    srOnly={false}
  >
    Name
     
    <span
      className="font-size-sm g600-text-clr ml-3"
    >
      Maximun 400 characters
    </span>
  </ForwardRef>
  <ForwardRef
    as="textarea"
    className="undefined rounded-0"
    maxLength={400}
  />
</ForwardRef>
`;

exports[`Components should render correctly for AllLinksButton 1`] = `
<button
  className="d-inline-flex align-items-center align-middle btn-upper dark-background text-white"
  onClick={[Function]}
>
  <img
    className="d-md-inline order-md-2 all-links-image"
  />
  <span
    className="font-size-sm m-3"
  >
    All Links
  </span>
</button>
`;

exports[`Components should render correctly for HambugerButton 1`] = `
<div
  className="mr-3 ml-3 position-relative hamburger "
  onClick={[Function]}
>
  <span
    className="position-absolute"
  />
  <span
    className="position-absolute"
  />
  <span
    className="position-absolute"
  />
  <span
    className="position-absolute"
  />
</div>
`;

exports[`Components should render correctly for NotificationsBadge 1`] = `
<button
  className="notifications-badge align-middle"
  onClick={[Function]}
>
  <img
    className="mr-2 d-inline"
    height={18}
    src="test-file-stub"
    width={15}
  />
  <ForwardRef
    className="ml-3 d-flex justify-content-center"
    pill={false}
    variant="danger"
  >
    3
  </ForwardRef>
</button>
`;

exports[`Components should render correctly for SiteLogo 1`] = `
<span
  className="site-logo cursor-pointer"
  onClick={[Function]}
  role="link"
  tabIndex={0}
>
  <ForwardRef
    fluid={false}
    rounded={false}
    roundedCircle={false}
    src="test-file-stub"
    thumbnail={false}
  />
</span>
`;

exports[`Components should render correctly for UserBadge 1`] = `
<span
  className="undefined user-badge text-white d-inline-flex align-items-center align-middle cursor-pointer text-decoration-none"
  onClick={[Function]}
  role="link"
  tabIndex={0}
>
  <span
    className="d-none d-sm-inline order-xl-2 font-weight-bold"
  >
    Luz Lacson
  </span>
  <ForwardRef
    alt="Luz Lacson"
    className="order-xl-1"
    fluid={false}
    rounded={false}
    roundedCircle={true}
    src="https://media.licdn.com/dms/image/C4D03AQHgxarEg1Cvuw/profile-displayphoto-shrink_800_800/0?e=1575504000&v=beta&t=thBaLX-J3-MoHdpJ3MvombNd4LOKfMHNRPLlmopM6KI"
    thumbnail={false}
  />
</span>
`;

exports[`should render correctly for Alert Modal 1`] = `
<ForwardRef(render)
  className="alert-modal"
  onHide={[Function]}
  show={true}
>
  <ForwardRef
    className="text-condensed font-weight-bold text-center pt-5"
  >
    Please enter both month and day for Birthday.
  </ForwardRef>
  <ForwardRef
    className="border-0 d-flex justify-content-center"
  >
    <ForwardRef
      className="w-100 mb-5"
      noGutters={false}
    >
      <ForwardRef
        className="text-center"
        lg="12"
        md="12"
      >
        <Unknown
          className="font-size-sm text-upper primary-button"
          handleClick={[Function]}
          title="Ok"
          uniqueIdentifier="close"
        />
      </ForwardRef>
    </ForwardRef>
  </ForwardRef>
</ForwardRef(render)>
`;

exports[`should render correctly for Button with Icon 1`] = `
<a
  className="font-size-sm cursor-pointer undefined"
  onClick={[Function]}
>
  <ForwardRef
    fluid={false}
    height={20}
    rounded={false}
    roundedCircle={false}
    src="test-file-stub"
    thumbnail={false}
    width={20}
  />
  <span
    className="pl-2 gray-600"
  >
    Add to My Contacts
  </span>
</a>
`;

exports[`should render correctly for CheckBox 1`] = `
<ForwardRef
  inline={false}
  onClick={[Function]}
>
  <ForwardRef
    custom={true}
    disabled={false}
    inline={false}
    isInvalid={false}
    isValid={false}
    label="CheckLabel"
    title=""
    type="checkbox"
  />
</ForwardRef>
`;

exports[`should render correctly for CheckBox 2`] = `
<div
  className="left-label-checkbox font-size-sm pr-1"
  onClick={[Function]}
>
  <label
    className="checked"
  >
    label
  </label>
  <input
    checked={true}
    type="checkbox"
  />
</div>
`;

exports[`should render correctly for Close toggle Button 1`] = `
<ForwardRef
  active={false}
  className="dark-background p-3 menu-toggler"
  disabled={false}
  onClick={[Function]}
  type="button"
  variant="dark"
>
  <img
    height={50}
    src="test-file-stub"
    width={50}
  />
</ForwardRef>
`;

exports[`should render correctly for Confirmation Modal 1`] = `
<ForwardRef(render)
  className="confirmation-modal"
  onHide={[Function]}
  show={true}
>
  <ForwardRef
    className="text-condensed font-weight-bold text-center pt-5"
  >
    Are you sure you want to remove this contact?
  </ForwardRef>
  <ForwardRef
    className="border-0 d-flex justify-content-center"
  >
    <ForwardRef
      className="w-100 mb-5"
      noGutters={false}
    >
      <ForwardRef
        className="text-center"
        xs="6"
      >
        <Unknown
          className="font-size-sm text-upper"
          handleClick={[Function]}
          title="Cancel"
          uniqueIdentifier="close"
        />
      </ForwardRef>
      <ForwardRef
        className="text-center"
        xs="6"
      >
        <Unknown
          className="font-size-sm text-upper"
          handleClick={[Function]}
          title="Remove"
          uniqueIdentifier="save"
        />
      </ForwardRef>
    </ForwardRef>
  </ForwardRef>
</ForwardRef(render)>
`;

exports[`should render correctly for Edit Button 1`] = `
<a
  className="edit-button cursor-pointer "
  href="https://www.google.com"
  onClick={[Function]}
>
  <ForwardRef
    className="pr-1 d-inline"
    fluid={false}
    height={15}
    rounded={false}
    roundedCircle={false}
    src="test-file-stub"
    thumbnail={false}
    width={15}
  />
  <span
    className="d-sm-inline order-xl-2 font-size-xs text-uppercase letter-spacing-01rem"
  >
    Edit
  </span>
</a>
`;

exports[`should render correctly for Error 1`] = `
<div
  className="text-center mt-5"
>
  <ForwardRef
    fluid={false}
    height={115}
    rounded={false}
    roundedCircle={false}
    src="test-file-stub"
    thumbnail={false}
    width={150}
  />
  <p
    className="mt-5 gray-600 font-weight-bold font-size-title"
  >
    Please upgrade your browser.
  </p>
  <p
    className="font-size-base m-auto align-text-center"
  >
    UMG Central has detected that you are using an unsupported browser. Please use links below to download and install a supported browser.
  </p>
  <div
    className="mt-5 d-flex justify-content-center"
  >
    <div
      className="mr-5"
    >
      <ForwardRef
        fluid={false}
        height={50}
        rounded={false}
        roundedCircle={false}
        thumbnail={false}
        width={50}
      />
      <a
        className="d-block font-size-sm text-decoration-underline"
        href="#"
      >
        Google Chrome
      </a>
    </div>
    <div
      className="mr-5"
    >
      <ForwardRef
        fluid={false}
        height={50}
        rounded={false}
        roundedCircle={false}
        thumbnail={false}
        width={50}
      />
      <a
        className="d-block font-size-sm text-decoration-underline"
        href="#"
      >
        Apple Safari
      </a>
    </div>
    <div>
      <ForwardRef
        fluid={false}
        height={50}
        rounded={false}
        roundedCircle={false}
        thumbnail={false}
        width={50}
      />
      <a
        className="d-block font-size-sm text-decoration-underline"
        href="#"
      >
        Microsoft Edge
      </a>
    </div>
  </div>
</div>
`;

exports[`should render correctly for Error Restricted 1`] = `
<div
  className="text-center mt-5"
>
  <ForwardRef
    fluid={false}
    height={115}
    rounded={false}
    roundedCircle={false}
    src="test-file-stub"
    thumbnail={false}
    width={150}
  />
  <p
    className="mt-5 gray-600 font-weight-bold font-size-title"
  >
    Restricted Content
  </p>
  <p
    className="font-size-base m-auto align-text-center"
  >
    To access this content, please connect to the UMG Network via VPN.
  </p>
  <div
    className="mt-5 d-flex justify-content-center"
  >
    <a
      className="text-upper gray-700 font-size-sm"
      href="https://central.umusic.com/pages/315"
    >
      <u
        className="letter-spacing"
      >
        View vpn instructions
      </u>
    </a>
  </div>
</div>
`;

exports[`should render correctly for FooterFeedbackView 1`] = `
<div
  className="footer-container w-100 text-sm-right text-center d-flex flex-column justify-content-center"
>
  <ForwardRef
    fluid={false}
  >
    <a
      className="d-sm-inline order-xl-2 btn-secondary btn-feedback font-weight-bold text-uppercase letter-spacing-01rem"
    >
      Edit
    </a>
  </ForwardRef>
</div>
`;

exports[`should render correctly for Organisation Chart 1`] = `
<ForwardRef
  body={false}
  className="cards-list-container fix-height p-3 border-0 rounded-0 h-100"
>
  <ForwardRef>
    <div
      className="d-flex justify-content-between align-items-baseline mb-3"
    >
      <ForwardRef
        className="text-upper mb-4 font-size-md"
      >
        ORG CHART
      </ForwardRef>
    </div>
    <div
      className="org-chart font-size-sm overflow-auto"
    >
      <ul>
        <li
          className="list-unstyled position-relative"
        >
          <button
            className="w-100"
            onClick={[Function]}
          >
            <ForwardRef
              className="org-chart-child-card text-left m-0 pl-3"
              fluid={false}
            >
              <ForwardRef
                className="pl-2 font-weight-bold pr-2"
                noGutters={false}
              >
                Tom Mitchell
              </ForwardRef>
              <ForwardRef
                className="pl-2 text-condensed"
                noGutters={false}
              >
                CEO
              </ForwardRef>
            </ForwardRef>
          </button>
          <ul>
            <li
              className="list-unstyled position-relative"
            >
              <div>
                <ForwardRef
                  className="org-chart-parent-card m-0  pt-2 pl-3 pb-2"
                  fluid={false}
                >
                  <ForwardRef
                    className="pl-2 font-weight-bold pr-2 text-left org-chart-parent-card-font"
                    noGutters={false}
                  >
                    Luz Lacson
                  </ForwardRef>
                </ForwardRef>
              </div>
              <ol>
                <li
                  className="list-unstyled position-relative"
                >
                  <button
                    className="w-100"
                    onClick={[Function]}
                  >
                    <ForwardRef
                      className="org-chart-child-card  text-left m-0 pl-3"
                      fluid={false}
                    >
                      <ForwardRef
                        className="pl-2 font-weight-bold pr-2"
                        noGutters={false}
                      >
                        Harman Kaur
                      </ForwardRef>
                      <ForwardRef
                        className="pl-2 text-condensed"
                        noGutters={false}
                      >
                        Business Analyst
                      </ForwardRef>
                    </ForwardRef>
                  </button>
                </li>
                <li
                  className="list-unstyled position-relative"
                >
                  <button
                    className="w-100"
                    onClick={[Function]}
                  >
                    <ForwardRef
                      className="org-chart-child-card  text-left m-0 pl-3"
                      fluid={false}
                    >
                      <ForwardRef
                        className="pl-2 font-weight-bold pr-2"
                        noGutters={false}
                      >
                        Ed Rowell
                      </ForwardRef>
                      <ForwardRef
                        className="pl-2 text-condensed"
                        noGutters={false}
                      >
                        Sr. Business Analyst
                      </ForwardRef>
                    </ForwardRef>
                  </button>
                </li>
              </ol>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </ForwardRef>
</ForwardRef>
`;

exports[`should render correctly for Partition with message 1`] = `
<div
  className="gray-600"
>
  <p
    className="text-upper font-size-md text-center top-gray-border"
  >
    The information below is private
  </p>
</div>
`;

exports[`should render correctly for Primary button 1`] = `
<ForwardRef
  active={false}
  className="btn-primary font-size-sm letter-spacing-01rem font-weight-bold text-uppercase undefined"
  disabled={false}
  onClick={[Function]}
  type="button"
  variant="primary"
>
  Primary Button
</ForwardRef>
`;

exports[`should render correctly for Secondary Button 1`] = `
<ForwardRef
  active={false}
  className="btn-secondary font-weight-bold text-uppercase letter-spacing-01rem"
  disabled={false}
  onClick={[Function]}
  type="button"
  variant="outline-secondary"
>
  <span
    className="d-sm-inline order-xl-2"
  >
    Secondary Button
  </span>
</ForwardRef>
`;

exports[`should render correctly for Thumb button like 1`] = `
<button
  className="like-icon d-flex align-items-center"
  onClick={[Function]}
>
  <ForwardRef
    fluid={false}
    height={14}
    rounded={false}
    roundedCircle={false}
    src="test-file-stub"
    thumbnail={false}
    width={14}
  />
  <span
    className="ml-1"
  >
    12
  </span>
</button>
`;

exports[`should render correctly for Thumb button unlike 1`] = `
<button
  className="unlike-icon d-flex align-items-center"
  onClick={[Function]}
>
  <ForwardRef
    fluid={false}
    height={14}
    rounded={false}
    roundedCircle={false}
    src="test-file-stub"
    thumbnail={false}
    width={14}
  />
  <span
    className="ml-1"
  >
    12
  </span>
</button>
`;

exports[`should render correctly for Toast 1`] = `
<div
  className="toast toast-success show"
>
  Added to My Contacts
</div>
`;

exports[`should render correctly for Toast 2`] = `
<div
  className="toast toast-error show"
>
  Contact is already in My Contacts
</div>
`;

exports[`should render correctly for Upload Button 1`] = `
<button
  className="upload-button font-weight-bold text-uppercase letter-spacing-01rem undefined"
  id="1"
  onClick={[Function]}
>
  Upload
</button>
`;
